### PR TITLE
Add tests for %TypedArray%.prototype.copyWithin

### PR DESCRIPTION
**Pull request by leobalter**

### CODE REVIEW

File: test/built-ins/TypedArray/prototype/copyWithin/return-this.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  performing a [[Get]] of "length" and the actual copying of values in step 12
+  must be performed in a manner that preserves the bit-level encoding of the
+  source data.
+
+  ...
+
+  22.1.3.3 Array.prototype.copyWithin (target, start [ , end ] )
+
+  13. Return O.
+includes: [compareArray.js, testTypedArray.js]
```
Comment (jugglinmike) [inline test/built-ins/TypedArray/prototype/copyWithin/return-this.js]: No need for "compareArray.js" in this test

**Comment by jugglinmike** [comment]
Looking good, Leo! Besides the one optimization I suggested above, could you normalize the file endings? Currently, the majority end with a single newline, but some end with multiple newlines, and others end without any newlines at all.

File: test/built-ins/TypedArray/prototype/copyWithin/return-this.js (hunk 1 above)
Comment (leobalter) [inline test/built-ins/TypedArray/prototype/copyWithin/return-this.js]: f1ece23

**Comment by leobalter** [comment]
Done!

**Comment by jugglinmike** [comment]
Thanks--this looks good to me!

**Comment by jugglinmike** [comment]
Rebased, squashed, and merged to `master` at commit 88f390d189cf59567e91b56e9993b001abcb3bfa. Thanks, Leo!

**Pull request closed without merging**